### PR TITLE
[C#][client][csharp-netcore] Fix csharp netcore defaultheaders

### LINKED ISSUE #3561: [BUG] [csharp-netcore] DefaultHeader in Configuration are ignored
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] <s>Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?</s> N/A
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

The `Configuration` of the ApiClient has a `DefaultHeader` property (which, tangentially, shoud be named `DefaultHeaders`) - an `IDictionary<string, string>`.

However, adding anything to this dictionary appears to have no effect. Those "default headers" are not sent with any request.

##### openapi-generator version

4.1.0-SNAPSHOT via Docker.

##### OpenAPI declaration file content or url

N/A - it occurs with any generated SDK

##### Command line used for generation

`generate -g csharp-netcore --additional-properties=targetFramework=netstandard2.0`

##### Steps to reproduce

1. Put something in DefaultHeaders, either in `GlobalConfiguration.Instance` or as a `Configuration` constructor parameter to an API object.
2. Make any request through the generated SDK.
3. Observe that default headers are not sent with any request.

##### Related issues/PRs

N/A

##### Suggest a fix

Analyzing the generated code, it turns out there is not a single reference to the `DefaultHeader` property in the context of a HTTP request being constructed or sent. Clearly, the code template has to be fixed.

### PR DESCRIPTION
**Pull request by mario-d-s**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@mandrean (2017/08), @jimschubert (2017/09)

### Description of the PR

Fix #3561

Previously, the .NET Core SDK would not send configured default headers. This PR fixes this and also "fixes" the strange singular name of the property by marking it obsolete and redirecting to the new correctly-named plural form.

This second change is a separate commit and is less important so I'd drop it if need be.

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

File: modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache [hunk 1]
```diff
@@ -94,7 +94,7 @@ namespace {{packageName}}.Client
         {
             UserAgent = "{{#httpUserAgent}}{{.}}{{/httpUserAgent}}{{^httpUserAgent}}OpenAPI-Generator/{{packageVersion}}/csharp{{/httpUserAgent}}";
             BasePath = "{{{basePath}}}";
-            DefaultHeader = new {{^net35}}Concurrent{{/net35}}Dictionary<string, string>();
+            DefaultHeaders = new {{^net35}}Concurrent{{/net35}}Dictionary<string, string>();
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache]: @mario-d-s I suggest we keep `DefaultHeader` as switching to `DefaultHeaders` will make this PR a breaking changes (without fallback), which means we will need to target this change to 5.0.x branch instead (scheduled to be released next year May)

Comment (mario-d-s) [reply modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache]: As a matter of fact I _do_ keep the `DefaultHeader` around and loop it through the new `DefaultHeaders` (both `get` and `set`), but marked it as obsolete ([see](#diff-a166220228f92ff881905c32eade1d60R159)). I subsequently adjusted the rest of the files from using the obsolete property to the new one.

That is not to say I might not have missed a place where this **is** a breaking change, in that case please let me know but it should be safe AFAIK.

**Comment by mario-d-s** [comment]
> Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.
> 
> Let me know if you need help fixing it.
> 
> Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

Oh thanks for spotting that! That's what you get when working from your day-job machine :sweat_smile:. Let's get that straightened up.

File: modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/csharp-netcore/Configuration.mustache]: Let's go with what you've and see if the community (users) spot anything we missed.

**Comment by wing328** [comment]
CircleCI failure has been fixed in the master.

**Comment by wing328** [comment]
@mario-d-s  thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**